entanglement_forging/core/wrappers/entanglement_forged_vqe.py

```diff
@@ -75,15 +107,52 @@ def __init__(self, ansatz, bitstrings, config, forged_operator, classical_energi
         self._paramsets_each_iteration = []
         self._schmidt_coeffs_each_iteration_each_paramset = []
         self._zero_noise_extrap = config.zero_noise_extrap
-        self.bitstrings = bitstrings
-        self._bitstrings_s = np.asarray(bitstrings)
+
+        self.bitstrings_u = bitstrings_u
+        self._bitstrings_s_u = np.asarray(bitstrings_u)
+
+        # Prevent unnecessary duplication of ansatz calculation if U=V
+        if bitstrings_v == bitstrings_u:
+            self.bitstrings_v = []
+        else:
+            self.bitstrings_v = bitstrings_v
+
+        # Make circuits which prepare states U|b_n_u> and U|phi^p_nm_u>
         (
-            self._tensor_prep_circuits,
-            self._superpos_prep_circuits,
-        ) = make_stateprep_circuits(bitstrings, config.fix_first_bitstring)
+            self._tensor_prep_circuits_u,
+            self._superpos_prep_circuits_u,
+            superpos_coeffs_u,
+        ) = make_stateprep_circuits(
+            bitstrings_u, config.fix_first_bitstring, suffix="u"
+        )
+
+        self._tensor_prep_circuits_v = []
+        self._superpos_prep_circuits_v = []
+        superpos_coeffs_v = []
+        self._bitstrings_s_v = np.array([])
+        if self.bitstrings_v:
+            # Make circuits which prepare states V|b_n_v> and V|phi^p_nm_v>
+            # Where U = V but bitstring |b_n_v> does not necessarily equal |b_n_u>
+            self._bitstrings_s_v = np.asarray(bitstrings_v)
+            (
+                self._tensor_prep_circuits_v,
+                self._superpos_prep_circuits_v,
+                superpos_coeffs_v,
+            ) = make_stateprep_circuits(
+                bitstrings_v, config.fix_first_bitstring, suffix="v"
+            )
+
+            # Offset the superpos coeffs associated with V so the indices match up
```

Comment:
Add a note in this comment saying that this constant factor of 2 will need to change when we support imaginary values, as the set P = {0, 2} from the equation will then be P = {0, 1, 2, 3}

Reply:
Just make the 2 a variable so it is more obvious